saltapi/repository/institution_repository.py

```diff
@@ -37,3 +41,107 @@ def get_institutions(self) -> List[Dict[str, Any]]:
             for row in result
         ]
         return institutions
+
+    def get_institution_by_name(self, institution_name: str) -> Dict[str, Any]:
```

Comment:
You assume that the institution name is unique. But I'd argue that it is the combination of institution name and department that should be considered unique. So you could only get the institution by name and department.